e2e/portal/tests/ProfilePageActivityLog/ViewTransactions.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { SeedScript } from '@121-service/src/scripts/enum/seed-script.enum';
+import NLRCProgram from '@121-service/src/seed-data/program/program-nlrc-pv.json';
+import { PermissionEnum } from '@121-service/src/user/enum/permission.enum';
+import {
+  createAndStartPayment,
+  waitForMessagesToComplete,
+  waitForPaymentTransactionsToComplete,
+} from '@121-service/test/helpers/program.helper';
+import {
+  getAllActivitiesCount,
+  seedIncludedRegistrations,
+  updateRegistration,
+} from '@121-service/test/helpers/registration.helper';
+import {
+  createUserWithPermissions,
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+import {
+  programIdPV,
+  registrationPV5,
+} from '@121-service/test/registrations/pagination/pagination-data';
+
+import TableComponent from '@121-e2e/portal/components/TableComponent';
+import LoginPage from '@121-e2e/portal/pages/LoginPage';
+import RegistrationActivityLogPage from '@121-e2e/portal/pages/RegistrationActivityLogPage';
+import RegistrationsPage from '@121-e2e/portal/pages/RegistrationsPage';
+
+const referenceIdPV5 = registrationPV5.referenceId;
+let activitiesCount: number;
+
+const loginAs = async ({ page, username }) => {
+  const loginPage = new LoginPage(page);
+  await page.goto(`/`);
+  await loginPage.login(username);
+};
+
+// Arrange
+test.beforeEach(async () => {
+  await resetDB(SeedScript.nlrcMultiple, __filename);
+
+  const accessToken = await getAccessToken();
+  await seedIncludedRegistrations([registrationPV5], programIdPV, accessToken);
+
+  const paymentResponse = await createAndStartPayment({
+    programId: 2,
+    transferValue: 100,
+    referenceIds: [referenceIdPV5],
+    accessToken,
+  });
+  const paymentId = paymentResponse.body.id;
+
+  await waitForPaymentTransactionsToComplete({
+    programId: programIdPV,
+    paymentId,
+    paymentReferenceIds: [referenceIdPV5],
+    accessToken,
+    maxWaitTimeMs: 2_000,
+    completeStatuses: [
+      TransactionStatusEnum.success,
+      TransactionStatusEnum.waiting,
+    ],
+  });
+
+  await updateRegistration(
+    2,
+    referenceIdPV5,
+    {
+      maxPayments: '2',
+    },
+    'automated test',
+    accessToken,
+  );
+
+  await waitForMessagesToComplete({
+    programId: programIdPV,
+    referenceIds: [referenceIdPV5],
+    accessToken,
+    expectedMessageAttribute: {
+      key: 'contentType',
+      values: [MessageContentType.paymentInstructions], // Payment instruction is the last message sent when doing a payment with voucher
+    },
+  });
+
+  activitiesCount = (
+    await getAllActivitiesCount(programIdPV, referenceIdPV5, accessToken)
+  ).totalCount;
+});
+
+test.describe('as admin user', () => {
+  test.beforeEach(async ({ page }) => {
+    await loginAs({
+      page,
+      username: env.USERCONFIG_121_SERVICE_EMAIL_ADMIN,
+    });
+  });
+
+  test('Expand rows of activity overview and view transaction history', async ({
```

Comment:
this is not in line with your explanation earlier that describe > test > test.step should read as a sentence, right?

Reply:
improved it a bit